hcl/writer.go

```diff
@@ -286,6 +288,16 @@ func walkVariables(cfg map[string]interface{}, validVariables map[string]struct{
 		case map[string]interface{}:
 			cfg[key] = walkVariables(v, validVariables, currentKey, variables)
 		case []interface{}:
+			if len(v) == 0 {
+				if hasKey(validVariables, currentKey) {
+					varName := strings.ReplaceAll(currentKey, ".", "_")
+					variables[varName] = map[string]interface{}{
+						"default": cfg[key],
+					}
+					cfg[key] = fmt.Sprintf("${var.%s}", varName)
```

Comment:
is it mandatory to keep this old notation `${...}` ?

Reply:
Yes, because then the formatter removes it, so from `"key": "${var.name}"` to `"key": var.name`.